apps/server/src/migrations/0234__multi_user_support.ts

```diff
@@ -0,0 +1,204 @@
+ * 4. Creates note_shares table for shared notes
+ * 5. Adds userId column to existing tables (notes, branches, options, etapi_tokens, etc.)
+ * 6. Creates a default admin user with existing password
+ * 7. Associates all existing data with the admin user
+ */
+
+import sql from "../services/sql.js";
+import optionService from "../services/options.js";
+import { randomSecureToken, toBase64 } from "../services/utils.js";
+import myScryptService from "../services/encryption/my_scrypt.js";
+
+export default async () => {
+    console.log("Starting multi-user support migration (v234)...");
+
+    // 1. Create users table
+    sql.execute(`
+        CREATE TABLE IF NOT EXISTS "users" (
+            userId TEXT PRIMARY KEY,
+            username TEXT NOT NULL UNIQUE,
+            email TEXT,
+            passwordHash TEXT NOT NULL,
+            passwordSalt TEXT NOT NULL,
+            derivedKeySalt TEXT NOT NULL,
+            encryptedDataKey TEXT,
+            isActive INTEGER NOT NULL DEFAULT 1,
+            isAdmin INTEGER NOT NULL DEFAULT 0,
+            utcDateCreated TEXT NOT NULL,
+            utcDateModified TEXT NOT NULL
+        )
+    `);
+
+    sql.execute(`CREATE INDEX IF NOT EXISTS IDX_users_username ON users (username)`);
+    sql.execute(`CREATE INDEX IF NOT EXISTS IDX_users_email ON users (email)`);
+    sql.execute(`CREATE INDEX IF NOT EXISTS IDX_users_isActive ON users (isActive)`);
+
+    // 2. Create roles table
+    sql.execute(`
+        CREATE TABLE IF NOT EXISTS "roles" (
+            roleId TEXT PRIMARY KEY,
+            name TEXT NOT NULL UNIQUE,
+            description TEXT,
+            permissions TEXT NOT NULL,
+            utcDateCreated TEXT NOT NULL,
+            utcDateModified TEXT NOT NULL
+        )
+    `);
+
+    // 3. Create user_roles junction table
+    sql.execute(`
+        CREATE TABLE IF NOT EXISTS "user_roles" (
+            userId TEXT NOT NULL,
+            roleId TEXT NOT NULL,
+            utcDateAssigned TEXT NOT NULL,
+            PRIMARY KEY (userId, roleId),
+            FOREIGN KEY (userId) REFERENCES users(userId) ON DELETE CASCADE,
+            FOREIGN KEY (roleId) REFERENCES roles(roleId) ON DELETE CASCADE
+        )
+    `);
+
+    // 4. Create note_shares table for sharing notes between users
+    sql.execute(`
+        CREATE TABLE IF NOT EXISTS "note_shares" (
+            shareId TEXT PRIMARY KEY,
+            noteId TEXT NOT NULL,
+            ownerId TEXT NOT NULL,
+            sharedWithUserId TEXT NOT NULL,
+            permission TEXT NOT NULL DEFAULT 'read',
+            utcDateCreated TEXT NOT NULL,
+            utcDateModified TEXT NOT NULL,
+            isDeleted INTEGER NOT NULL DEFAULT 0,
+            FOREIGN KEY (noteId) REFERENCES notes(noteId) ON DELETE CASCADE,
+            FOREIGN KEY (ownerId) REFERENCES users(userId) ON DELETE CASCADE,
+            FOREIGN KEY (sharedWithUserId) REFERENCES users(userId) ON DELETE CASCADE
+        )
+    `);
```

Comment:
If the tables are not used yet (e.g. in order to be able to share notes), there is no reason to create them.